.github/actions_scripts/getChangelog.pl

```diff
@@ -2,18 +2,34 @@
 
 use strict;
 use warnings;
-# get the file path
-my $ChangeLogFile = $ARGV[0];
-# get the version
-my $Version = quotemeta $ARGV[1];
-# open the file (slurp mode): https://perlmaven.com/slurp
+# get arguments
+my ($ChangeLogFile, $Version, $FormatOption) = @ARGV;
+# open the file
 open my $fh, '<', $ChangeLogFile or die;
-$/ = undef;
-my $ChangeLog = <$fh>;
-close $fh;
 
-if ( $ChangeLog =~ m/^### $Version (.*?)$(.*?)^###/sm ) {
-    print $2;
-} else {
-    print "No changelog found for this version.";
+my $SingleLineEntries = ($FormatOption or '') eq '--line-per-entry';
+
+my $logversion = "";
+my $entry;
+while (my $line = <$fh>) {
```

Comment:
I *think* we should stop looping through the file if we already found the correct version ($logversion changed to the next version). Then we could also check if a version wasn't found as soon as we reached the EOF and output some error message.

Reply:
@npostavs could you please insert a break statement to stop looping through unneeded lines?

Reply:
Oh yep, done.

Reply:
Thanks!